### PR TITLE
delete sprint 1 and obsticalAvoidance

### PR DESCRIPTION
**Pull request by LIYUHAN22**
delete sprint 1 and obsticalAvoidance from master

### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Code change tested manually if not covered by automated tests
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change

### CODE REVIEW

**Comment by LIYUHAN22** [comment]
publisButton of all the buttons, and part of the stream.

File: arduino/smartcar/smartcar.ino [hunk 1]
```diff
@@ -1,9 +1,264 @@
...
+        if(front.getDistance() < BrakeDistance && front.getDistance() > 0)
+        {
+            rotate(degreesToTurn, speedToTurn);
+            car.setSpeed(cruiseSpeed);
+            car.setAngle(0);
+            car.update();
+        }
+    }
+
+    if((front.getDistance() > 0 && front.getDistance() < crackDistance) && car.getSpeed() < 0.01 && cruiseFlag)
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino:254]: I am pretty sure we have talked about calling getDistance multiple times and why it's wrong.

File: arduino/smartcar/smartcar.ino [hunk 2]
```diff
@@ -1,9 +1,264 @@
...
+    }
+
+    if(car.getSpeed() != 0 && forward) 
+    {
+        obstacleAvoidance(BrakeDistance);
+    }
+
+    if(car.getSpeed() == 0 && cruiseFlag) 
+    {
+        if(front.getDistance() < BrakeDistance && front.getDistance() > 0)
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino:245]: Same problem as below

File: arduino/smartcar/smartcar.ino [hunk 3]
```diff
@@ -1,9 +1,264 @@
...
+    if((front.getDistance() > 0 && front.getDistance() < crackDistance) && car.getSpeed() < 0.01 && cruiseFlag)
+    {
+        Serial.println("go back");
+        go(backDistance, cruiseSpeed);
+        rotate(degreesToTurn, speedToTurn);
+        car.setSpeed(cruiseSpeed);
+        car.setAngle(0);
+        car.update();
+    }
+    delay(10);
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino:263]: Why do you delay for 10 milliseconds?

File: arduino/smartcar/smartcar.ino [hunk 4]
```diff
@@ -1,9 +1,264 @@
...
+        char input = Serial.read(); // read everything that has been received so far and log down
+                                    // the last entry
+        ctrlHeading(input);
+        if(input == 'c')
+        {
+            cruiseControl();
+        }
+    }
+
+    if(updateFlag)
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino:233]: What is the point of the update flag? Nothing bad happens if you call update and not in cruise control mode.

File: .gitignore [hunk 5]
```diff
@@ -2,6 +2,7 @@
 
 # Prerequisites
 *.d
+*.DS_Store
```
Comment (platisd) [inline .gitignore:5]: You are correctly ignoring this file, but at the same time you're adding it in the same pull request.

**Review by platisd — CHANGES_REQUESTED** [review]

**Pull request closed without merging**